src/reporting/market.se

```diff
@@ -580,39 +573,6 @@ def getAutomatedReportDisputeDueTimestamp():
     assertNoValue()
     return(self.getAutomatedReportDueTimestamp() + AUTOMATED_REPORTING_DISPUTE_DURATION_SECONDS)
 
-
-####
-#### Private
-####
-
-# FIXME: update to new mechanics and figure out a way to extract bond holder payments before redeeming bonds, current code will payout bond holders before extracting losing REP which means bond holders really only get 1x their bond at the moment
-def privateRewardBondHolders():
-    assertPrivateCall()
-    # require(self.isFinalized())
-    # availableRep = self.getReputationToken().balanceOf(self)
-    # desiredBondPot = self.getDesiredDisputeRewardPot()
-    # leftoverRep = availableRep - desiredBondPot
-    # # send any leftover REP to the reporting contract for distribution to REP holders
-    # if (leftoverRep > 0):
-    #     self.getReputationToken().transfer(self.getFinalWinningReportingToken(), leftoverRep)
-    # # if our pot doesn't have enough REP to cover bond holders, then take some REP holders
-    # elif (leftoverRep < 0):
-    #     self.getFinalWinningReportingToken().extractBondHolderPayments()
-    # automatedReporterPayoutPortion = 0
-    # limitedReportersPayoutPortion = 0
-    # allReportersPayoutPortion = 0
-    # if (self.automatedReporterPayoutDistributionHash != self.finalPayoutDistributionHash && self.automatedReporterDisputeBondHolder):
-    #     automatedReporterPayoutPortion = AUTOMATED_REPORTER_DISPUTE_BOND_AMOUNT
-    # if (self.limitedReportersPayoutDistributionHash != finalPayoutDistributionHash && self.limitedReportersDisputeBondHolder):
-    #     limitedReportersPayoutPortion = LIMITED_REPORTERS_DISPUTE_BOND_AMOUNT
-    # if (self.allReportersPayoutDistributionHash != finalPayoutDistributionHash && self.allReportersDisputeBondHolder):
-    #     allReportersPayoutPortion = ALL_REPORTERS_DISPUTE_BOND_AMOUNT
-    # automatedReporterBondPayout = availableRep * automatedReporterPayoutPortion / (automatedReporterPayoutPortion + limitedReportersPayoutPortion + allReportersPayoutPortion)
-    # limitedReportersBondPayout = (availableRep - automatedReporterBondPayout) * limitedReportersPayoutPortion / (limitedReportersPayoutPortion + allReportersPayoutPortion)
-    # allReportersBondPayout = availableRep - automatedReporterBondPayout - limitedReportersBondPayout
-    # if (automatedReporterBondPayout):
-    #     self.getReputationToken().transfer(self.automatedReporterDisputeBondHolder, automatedReporterBondPayout)
-    # if (limitedReportersBondPayout):
-    #     self.getReputationToken().transfer(self.limitedReportersDisputeBondHolder, limitedReportersBondPayout)
-    # if (allReportersBondPayout):
-    #     self.getReputationToken().transfer(self.allReportersDisputeBondHolder, allReportersBondPayout)
+def receiveToken():
+    if (msg.sender == self.getReputationToken())
+        #treat it like a dispute bond receipt.
```

Comment:
This won't compile as is.  Recommend deleting for now until we decide whether or not to move forward with the whole "receiveToken" pattern or not.